### PR TITLE
About and Activities UI refinement

### PR DESCRIPTION
**Pull request by comzeradd**
We no longer have Clubs roles so no need to hard code them in the template. I changed this block to be a `InputTextContentBlock` so now it's dynamic. The background image on the body block is also dynamic now.

This also containers some css adjustments to align with [Phase 1 design](https://github.com/mozilla/coss/issues/322).

### CODE REVIEW

**Comment by comzeradd** [comment]
I added a commit with UI adjustments to the Activities to meet the Phase 1 Design.

File: coss/home/static/js/clubs_about.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+$(document).ready(function() {
```
Comment (akatsoulas) [inline coss/home/static/js/clubs_about.js]: Why is this moved in the `home` app?

Comment (comzeradd) [reply coss/home/static/js/clubs_about.js]: All the javascript code there is needed for the parts of the template that live in the home app. I'll rename it to a more generic name.

**Comment by comzeradd** [comment]
PR updated

File: coss/home/models.py [hunk 2]
```diff
@@ -22,10 +22,18 @@ class AboutSnippet(models.Model):
...
+    body_image = models.ForeignKey(
+        'wagtailimages.Image',
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name='+'
+    )
     info_title = models.CharField(blank=True, default='', max_length=40)
     info_text = fields.RichTextField(blank=True)
+    details_title = models.CharField(blank=True, default='', max_length=40)
```
Comment (akatsoulas) [inline coss/home/models.py]: There is another `details` section in the streamfield? Are the two related? If yes, this shouldn't part of the snippet page. If not, it's a bit confusing to have details in two fields.

Comment (comzeradd) [reply coss/home/models.py]: Yes, good point. Since the rest of the details block is also generic and not specific to OS Clubs, I moved it inside that Snippet. Now it's less confusing for a Content creator.

screenshot-2017-9-6 wagtail - editing about snippet - test

File: coss/home/templates/home/about_page.html [hunk 3]
```diff
@@ -15,46 +15,80 @@
...
+        <a class="nav-link" href="#details">{{ page.about.details_title }}</a>
+      {% endif %}
     </nav>
 
     {% if page.about.body_text %}
-      <div id="body" class="body">
+      {% if page.about.body_image %}
+        {% image page.about.body_image original as bg_photo %}
+      {% endif %}
+      <div id="body" class="body" data-bg-img="{{ bg_photo.url }}">
```
Comment (akatsoulas) [inline coss/home/templates/home/about_page.html]: What happens if the `bg_photo` is undefined?

File: coss/home/templates/home/about_page.html [hunk 4]
```diff
@@ -15,46 +15,80 @@
...
+        <div class="body-image"></div>
         <div class="well">
+          {% if page.about.body_title %}
```
Comment (akatsoulas) [inline coss/home/templates/home/about_page.html]: Shouldn't this `if` be before the `<div class="well"`

File: coss/opensource_clubs/templates/opensource_clubs/about_page.html [hunk 5]
```diff
@@ -10,45 +10,8 @@
...
-            {{ page.roles_coordinator|richtext }}
-          </div>
-
-          <div class="nav-content" id="role-member">
-            {{ page.roles_member|richtext }}
-          </div>
-        </div>
-      </div>
-    </div>
+  <div class="collaborators">
```
Comment (akatsoulas) [inline coss/opensource_clubs/templates/opensource_clubs/about_page.html]: Since we have the snippet/streamfield for activities, don't we want to add the missing section in this PR?

File: coss/home/templates/home/about_page.html (hunk 3 above)
Comment (comzeradd) [reply coss/home/templates/home/about_page.html]: In that case that data attribute is empty and the javascript code will fallback to the default one.

File: coss/home/templates/home/about_page.html (hunk 4 above)
Comment (comzeradd) [reply coss/home/templates/home/about_page.html]: Not necessarily. We won't to display the well and `page.about.body_text` even when no title is set.

**Comment by comzeradd** [comment]
PR updated. I added the collaborators section. I also modified the model to group relevant fields. This will make it easier for Content creators to understand the scope of each field.

**Comment by comzeradd** [comment]
> Found another file with the destination path 'css/_colors.scss'. It will be ignored since only the first encountered file is collected. If this is not what you want, make sure every static file has a unique path.

That doesn't cause any problem. Sass ignores partial files anyway (the ones the begin with _), so it's ok to have the same name on different apps.

**Comment by akatsoulas** [comment]
Instead of ignoring it, can't we just fix it?

File: coss/home/models.py [hunk 6]
```diff
@@ -20,36 +22,80 @@ class Meta:
...
-    info_title = models.CharField(blank=True, default='', max_length=40)
-    info_text = fields.RichTextField(blank=True)
-    bottom_cta_title = models.CharField('Title', blank=True, default='', max_length=40)
-    bottom_cta_text = models.CharField('Text', blank=True, default='', max_length=40)
-    bottom_cta1_text = models.CharField('Text', blank=True, default='', max_length=20)
-    bottom_cta1_link = models.URLField('Link', blank=True, default='')
-    bottom_cta2_text = models.CharField('Text', blank=True, default='', max_length=20)
-    bottom_cta2_link = models.URLField('Link', blank=True, default='')
+    body_title = models.CharField(verbose_name='Title', blank=True, default='',
+                                  max_length=40)
```
Comment (akatsoulas) [inline coss/home/models.py]: Let's be consistent, in other places the title holds `100 chars`.

Comment (akatsoulas) [reply coss/home/models.py]: The above applies to all the title fields that exist in this PR.

File: coss/home/models.py [hunk 7]
```diff
@@ -20,36 +22,80 @@ class Meta:
...
+    details_title = models.CharField(verbose_name='Title', blank=True, default='',
+                                     max_length=40)
+    details = fields.StreamField([
+        ('section', InputTextContentBlock(),),
+    ], null=True, blank=True)
+    collaborators_title = models.CharField(verbose_name='Title', blank=True, default='',
+                                           max_length=40)
+    collaborators_text = fields.RichTextField(verbose_name='Text', blank=True)
+    collaborators_cta_text = models.CharField(verbose_name='CTA Text', blank=True, default='',
+                                              max_length=20)
```
Comment (akatsoulas) [inline coss/home/models.py]: `20` feels too restrictive here. maybe doubling that number?

**Comment by comzeradd** [comment]
Yes I'll make some tests. It seems to be a [known issue](https://github.com/django-compressor/django-compressor/issues/720). Although this is not related to this PR. This warning was introduced by 229efaea1fe27ca534405d1ac2991b26c5c9765d.

File: coss/home/templates/home/about_page.html (hunk 4 above)
Comment (akatsoulas) [reply coss/home/templates/home/about_page.html]: Agreed, but the comment stands for the body_text too. It's a not required field. Ideally, as we make the project more generic, we want to only render blocks that have content. In the future, this can be a standalone snippet inserted in a template.
Probably, it's OK for now, but please add at least comments as placeholders on what we need to revisit in order to make the template more generic.

File: coss/home/models.py [hunk 8]
```diff
@@ -20,36 +22,80 @@ class Meta:
...
-    bottom_cta_text = models.CharField('Text', blank=True, default='', max_length=40)
-    bottom_cta1_text = models.CharField('Text', blank=True, default='', max_length=20)
-    bottom_cta1_link = models.URLField('Link', blank=True, default='')
-    bottom_cta2_text = models.CharField('Text', blank=True, default='', max_length=20)
-    bottom_cta2_link = models.URLField('Link', blank=True, default='')
+    body_title = models.CharField(verbose_name='Title', blank=True, default='',
+                                  max_length=40)
+    body_text = fields.RichTextField(verbose_name='Text', blank=True)
+    body_image = models.ForeignKey(
+        'wagtailimages.Image',
```
Comment (akatsoulas) [inline coss/home/models.py]: Locally, the image that I chose is not displayed in the template. Please make sure that this is working properly before merging this PR. Also that the fallback is working too.

File: coss/home/models.py [hunk 9]
```diff
@@ -20,36 +22,80 @@ class Meta:
...
+        on_delete=models.SET_NULL,
+        related_name='+',
+        verbose_name='Image'
+    )
+    info_title = models.CharField(verbose_name='Title', blank=True, default='',
+                                  max_length=40)
+    info_text = fields.RichTextField(verbose_name='Text', blank=True)
+    details_title = models.CharField(verbose_name='Title', blank=True, default='',
+                                     max_length=40)
+    details = fields.StreamField([
```
Comment (akatsoulas) [inline coss/home/models.py]: We need to show in the snippet that details is a required field In the admin the user can only see the streamfield item and there is no indication that it's a required field.

File: coss/home/templates/home/about_page.html (hunk 4 above)
Comment (comzeradd) [reply coss/home/templates/home/about_page.html]: Yes, adding comments is a good idea. Just to point out that this whole block is wrapped inside an if statement (`{% if page.about.body_text %}`), so this is rendered only if a body text is provided.

File: coss/home/models.py (hunk 6 above)
Comment (comzeradd) [reply coss/home/models.py]: All `title` fields have the same limit. All `text` fields are either unlimited or are limited based on design (eg. buttons can only have really short text).

File: coss/home/models.py (hunk 9 above)
Comment (comzeradd) [reply coss/home/models.py]: Details is not required. We only render this block if we have at least on details block.
